public/expressions/overlay_anomalies.ts

```diff
@@ -0,0 +1,193 @@
+// /*
+//  * SPDX-License-Identifier: Apache-2.0
+//  *
+//  * The OpenSearch Contributors require contributions made to
+//  * this file be licensed under the Apache-2.0 license or a
+//  * compatible open source license.
+//  *
+//  * Any modifications Copyright OpenSearch Contributors. See
+//  * GitHub history for details.
+//  */
+
+import { get } from 'lodash';
+import { i18n } from '@osd/i18n';
+import { ExpressionFunctionDefinition } from '../../../../src/plugins/expressions/public';
+import {
+  VisLayerTypes,
+  VisLayers,
+  ExprVisLayers,
+  PluginResource,
+} from '../../../../src/plugins/vis_augmenter/public';
+import {
+  TimeRange,
+  calculateBounds,
+} from '../../../../src/plugins/data/common';
+import {
+  getAnomalySummaryQuery,
+  parsePureAnomalies,
+} from '../pages/utils/anomalyResultUtils';
+import { AD_NODE_API } from '../../utils/constants';
+import { AnomalyData } from '../models/interfaces';
+import { getClient } from '../services';
+import {
+  PointInTimeEventsVisLayer,
+  VisLayerError,
+  VisLayerErrorTypes,
+} from '../../../../src/plugins/vis_augmenter/public';
+import { PLUGIN_NAME } from '../../public/utils/constants';
+import { NO_PERMISSIONS_KEY_WORD } from '../../server/utils/helpers';
+
+type Input = ExprVisLayers;
+type Output = Promise<ExprVisLayers>;
+
+interface Arguments {
+  detectorId: string;
+}
+
+const name = 'overlay_anomalies';
+
+export type OverlayAnomaliesExpressionFunctionDefinition =
+  ExpressionFunctionDefinition<'overlay_anomalies', Input, Arguments, Output>;
+
+// This gets all the needed anomalies for the given detector ID and time range
+const getAnomalies = async (
+  detectorId: string,
+  startTime: number,
+  endTime: number
+): Promise<AnomalyData[]> => {
+  const anomalySummaryQuery = getAnomalySummaryQuery(
+    startTime,
+    endTime,
+    detectorId,
+    undefined,
+    false
+  );
+
+  // We set the http client in the plugin.ts setup() fn. We pull it in here to make a
+  // server-side call directly.
+  // Note we can't use the redux fns here (e.g., searchResults()) since it requires
+  // hooks (e.g., useDispatch()) which doesn't make sense in this context, plus is not allowed by React.
+  const anomalySummaryResponse = await getClient().post(
+    `..${AD_NODE_API.DETECTOR}/results/_search`,
+    {
+      body: JSON.stringify(anomalySummaryQuery),
+    }
+  );
+
+  return parsePureAnomalies(anomalySummaryResponse);
+};
+
+// This takes anomalies and returns them as vis layer of type PointInTimeEvents
+const convertAnomaliesToLayer = (
+  anomalies: AnomalyData[],
+  ADPluginResource: PluginResource
+): PointInTimeEventsVisLayer => {
+  const events = anomalies.map((anomaly: AnomalyData) => {
+    return {
+      timestamp: anomaly.startTime + (anomaly.endTime - anomaly.startTime) / 2,
+      metadata: {},
+    };
+  });
+  return {
+    originPlugin: 'anomaly-detection',
```

Comment:
I wonder if we should extract this into a constant. It might make it easier to track when processing the values on OSD side.

Reply:
Here also I wasn't 100% sure if it should be `anomaly-detection` or `anomaly-detection-dashboards` or even `anomalyDetectionDashboards` as we list on opensearch_dashboards.json. Definitely agree it should be a constant

Reply:
The main purpose of this field is to provide another way to categorize the `VisLayer`s. Few advantages include:
1. Plugins having the opportunity to filter all `VisLayer`s that match the plugin's `originPlugin` field for a particular embeddable to post-process the data, instead of having to re-fetch saved objects (not sure if applicable given the current way plugins are consuming visualize_embeddable, but it leaves that option open)
2. More categorizing/filtering within the view events page or other OSD-owned components that are displaying the `VisLayer`s in some organized manner

Perhaps these `originPlugin` constants, and `type` constants can be put in a dedicated constants file specific to `VisLayer`s / this feature, so we can kind of split up constants that are only used within our plugin vs. constants that may be utilized in OSD via the render framework.

Reply:
To clarify, this field isn't intended to be user-facing so it can be something internal like `anomalyDetectionDashboards` (this one makes sense to me to be consistent with the internal plugin name)

Reply:
I guess though, do you want the originPlugin constants here for the `VisLayer` interface and the name for the `PluginResource` be exactly the same?

Reply:
>  I guess this could be good to point out either in comments above the interface or to add to a plugin onboarding documentation of some sort

Yeah, that will be captured as part of https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3383